src/api/HttpError.js

```diff
@@ -0,0 +1,48 @@
+/* @flow */
+
+export type JsonApiErrorPayload = {
+  detail?: string,
+  code: string,
+  source?: {
+    pointer: string,
+  },
+  meta: any,
+};
+
+export default class HttpError extends Error {
+  response: Response;
+  status: number;
+  errors: Array<JsonApiErrorPayload> = [];
+  body: ?string = null;
+  extensions: ?{
+    upstream: {
+      status: number,
+      errors: JsonApiErrorPayload[],
+    },
+  };
+
+  constructor(response: Response) {
+    super();
+
+    this.response = response;
+    this.status = response.status;
+  }
+
+  async init() {
+    try {
+      this.body = await this.response.text();
+      this.errors = JSON.parse(this.body).errors;
+
+      this.message = this.errors && this.errors[0] && this.errors[0].detail;
+
+      this.extensions = {
+        upstream: {
+          status: this.status,
+          errors: this.errors,
```

Comment:
gonna be interesting to sync this up with payload keys w/the whole translation bit

Reply:
what is this bit even?

Reply:
it lets us pass through the HTTP errors so consumers can key off these, without having to use `formatError`